continuous_integration/scripts/install.sh

```diff
@@ -28,8 +28,13 @@ if [[ ${UPSTREAM_DEV} ]]; then
 fi
 
 # Install dask
+python -m pip install pytest-timeout
 python -m pip install --quiet --no-deps -e .[complete]
 echo conda list
 conda list
 
+# For debugging
+echo -e "--\n--Conda Environment (re-create this with \`conda env create --name <name> -f <output_file>\`)\n--"
+conda env export | grep -E -v '^prefix:.*$'
```

Comment:
Using the output of this to create a local conda environment, I'm able to reproduce the hanging behavior locally. This is producing some interesting output. For example, `pytest -sv dask/bag/tests/test_bag.py::test_groupby` has the following output 

```
Fatal Python error: deallocating None
Python runtime state: initialized

Current thread 0x0000000115312e00 (most recent call first):
  File "/Users/james/projects/dask/dask/dask/bag/core.py", line 1668 in partition
  File "/Users/james/projects/dask/dask/dask/core.py", line 121 in _execute_task
  File "/Users/james/projects/dask/dask/dask/local.py", line 222 in execute_task
  File "/Users/james/miniforge3/envs/dask-hanging-ci/lib/python3.8/multiprocessing/pool.py", line 125 in worker
  File "/Users/james/miniforge3/envs/dask-hanging-ci/lib/python3.8/multiprocessing/process.py", line 108 in run
  File "/Users/james/miniforge3/envs/dask-hanging-ci/lib/python3.8/multiprocessing/process.py", line 315 in _bootstrap
  File "/Users/james/miniforge3/envs/dask-hanging-ci/lib/python3.8/multiprocessing/spawn.py", line 129 in _main
  File "/Users/james/minifoFatal Python error: rge3/endeallocating Nonev
s/dasPython runtime state: kinitialized-han
ging-c
i/Current thread 0xlib/00000001056dee00p (most recent call first):
y  File th"o/nU3s.e8r/sm/ujlatiprocessing/spawn.pmy", line e116s in /sppraowjne_cmtasi/nd
a  File s"<string>", line 1 in <module>
k/dask/dask/bag/core.py", line 1668 in partition
  File "/Users/james/projects/dask/dask/dask/core.py", line 121 in _execute_task
  File "/Users/james/projects/dask/dask/dask/local.py", line 222 in execute_task
  File "/Users/james/miniforge3/envs/dask-hanging-ci/lib/python3.8/multiprocessing/pool.py", line 125 in worker
  File "/Users/james/miniforge3/envs/dask-hanging-ci/lib/python3.8/multiprocessing/process.py", line 108 in run
  File "/Users/james/miniforge3/envs/dask-hanging-ci/lib/python3.8/multiprocessing/process.py", line 315 in _bootstrap
  File "/Users/james/miniforge3/envs/dask-hanging-ci/lib/python3.8/multiprocessing/spawn.py", line 129 in _main
  File "/Users/james/miniforge3/envs/dask-hanging-ci/lib/python3.8/multiprocessing/spawn.py", line 116 in spawn_main
  File "<string>", line 1 in <module>
Fatal Python error: deallocating None
Python runtime state: initialized

Current thread 0x0000000114176e00 (most recent call first):
  File "/Users/james/projects/dask/dask/dask/bag/core.py", line 1668 in partition
  File "/Users/james/projects/dask/dask/dask/core.py", line 121 in _execute_task
  File "/Users/james/projects/dask/dask/dask/local.py", line 222 in execute_task
  File "/Users/james/miniforge3/envs/dask-hanging-ci/lib/python3.8/multiprocessing/pool.py", line 125 in worker
  File "/Users/james/miniforge3/envs/dask-hanging-ci/lib/python3.8/multiprocessing/process.py", line 108 in run
  File "/Users/james/miniforge3/envs/dask-hanging-ci/lib/python3.8/multiprocessing/process.py", line 315 in _bootstrap
  File "/Users/james/miniforge3/envs/dask-hanging-ci/lib/python3.8/multiprocessing/spawn.py", line 129 in _main
  File "/Users/james/miniforge3/envs/dask-hanging-ci/lib/python3.8/multiprocessing/spawn.py", line 116 in spawn_main
  File "<string>", line 1 in <module>
```